comp/dogstatsd/serverDebug/component.go

```diff
@@ -0,0 +1,49 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package serverDebug implements a component to run the dogstatsd server debug
+package serverDebug
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// team: agent-metrics-logs
+
+// Component is the component type.
+type Component interface {
+
+	// StoreMetricStats stores stats on the given metric sample.
+	StoreMetricStats(sample metrics.MetricSample)
+
+	// IsDebugEnabled gets the DsdServerDebug instance which provides metric stats
+	IsDebugEnabled() bool
+
+	// EnableMetricsStats enables metric stats tracking
+	EnableMetricsStats()
+
+	// DisableMetricsStats disables metric stats tracking
+	DisableMetricsStats()
```

Comment:
Note: You can have a single method for both `EnableMetricsStats` and `DisableMetricsStats`